hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java

```diff
@@ -0,0 +1,171 @@
+
+        this.accountState = states.get(TokenServiceImpl.ACCOUNTS_KEY);
+        this.aliases = states.get(TokenServiceImpl.ALIASES_KEY);
+    }
+
+    /**
+     * Persists a new {@link Account} into the state, as well as exporting its ID to the transaction
+     * receipt.
+     *
+     * @param account - the account to be added to modifications in state.
+     */
+    public void put(@NonNull final Account account) {
+        Objects.requireNonNull(account);
+        accountState.put(EntityNumVirtualKey.fromLong(account.accountNumber()), Objects.requireNonNull(account));
+    }
+
+    /**
+     * Persists a new alias linked to the account persisted to state
+     *
+     * @param alias - the alias to be added to modifications in state.
+     * @param accountNum - the account number to be added to modifications in state.
+     */
+    public void putAlias(@NonNull final String alias, @NonNull final long accountNum) {
+        Objects.requireNonNull(alias);
+        aliases.put(alias, new EntityNumValue(accountNum));
+    }
+
+    /** Commits the changes to the underlying data storage. */
+    public void commit() {
+        ((WritableKVStateBase) accountState).commit();
+        ((WritableKVStateBase) aliases).commit();
+    }
+
+    /**
+     * Returns the {@link Account} with the given number. If no such account exists, returns {@code
+     * Optional.empty()}
+     *
+     * @param accountID - the id of the Account to be retrieved.
+     */
+    @NonNull
+    public Optional<Account> get(final AccountID accountID) {
+        requireNonNull(accountID);
+        final var account = getAccountLeaf(accountID);
+        return Optional.ofNullable(account);
+    }
+
+    /**
+     * Returns the {@link Account} with the given number using {@link
+     * WritableKVState#getForModify(Comparable K)}. If no such account exists, returns {@code
+     * Optional.empty()}
+     *
+     * @param id - the number of the account to be retrieved.
+     */
+    @NonNull
+    public Optional<Account> getForModify(final AccountID id) {
+        // Get the account number based on the account identifier. It may be null.
+        final var accountOneOf = id.account();
+        final Long accountNum =
+                switch (accountOneOf.kind()) {
+                    case ACCOUNT_NUM -> accountOneOf.as();
+                    case ALIAS -> {
+                        final Bytes alias = accountOneOf.as();
+                        if (alias.length() == EVM_ADDRESS_LEN && isMirror(alias)) {
+                            yield fromMirror(alias);
+                        } else {
+                            final var entityNum = aliases.getForModify(alias.asUtf8String());
+                            yield entityNum == null ? EntityNumValue.DEFAULT.num() : entityNum.num();
+                        }
+                    }
+                    case UNSET -> EntityNumValue.DEFAULT.num();
+                };
+
+        return accountNum == null
```

Comment:
I don't think this can be `null`, since `fromMirror()` should never return `null` and the other branches of the `switch` above clearly don't.

Reply:
yeah removed.